Add extension guides to terraform provider (#590)
nitrictech/docs: src/pages/guides/terraform.mdx

import { SiTerraform } from 'react-icons/si'

export const description = 'How to work with the Terraform Providers'

# Terraform Guides

<GuidesGrid title="Recommended Terraform Guides">
    <Guide
    customIcon={SiTerraform}
    href="/guides/terraform/s3-encryption"
    name="AWS S3 Server-side Encryption"

>     description="Extend the Nitric AWS S3 Terraform provider to add Server-side encryption"

Reviewer: ```suggestion
    description="Extend the Nitric AWS S3 Terraform provider to add server-side encryption"
```

/>

  <Guide
    customIcon={SiTerraform}
    href="/guides/terraform/s3-replicate"
    name="AWS S3 Replication"

>     description="Extend the Nitric AWS S3 Terraform provider to replicate your S3 bucket"

Reviewer: ```suggestion
    description="Extend the Nitric AWS S3 Terraform provider to replicate S3 buckets"
```

/>

  <Guide
    customIcon={SiTerraform}
    href="/guides/terraform/api-gateway-throttle"
    name="AWS API Gateway Throttle"

>     description="Extend the Nitric API Gateway Terraform provider to set throttling limits"

Reviewer: ```suggestion
    description="Extend the Nitric AWS Terraform provider to set API Gateway throttling limits"
```
